library/alloc/src/collections/binary_heap.rs

```diff
@@ -708,6 +710,93 @@ impl<T: Ord> BinaryHeap<T> {
         self.data.retain(f);
         self.rebuild();
     }
+
+    /// Returns `true` if the `BinaryHeap` contains the specified item.
+    ///
+    /// The item may be any borrowed form of the heap's item type, but
+    /// `PartialEq` on the borrowed form *must* match that for the item type.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// #![feature(binary_heap_contains_remove)]
+    /// use std::collections::BinaryHeap;
+    ///
+    /// let heap = BinaryHeap::from(vec![1, 3, 5, 7, 9]);
+    ///
+    /// assert!(heap.contains(&5));
+    /// assert!(!heap.contains(&2));
+    /// ```
+    ///
+    /// # Time complexity
+    ///
+    /// The worst case time complexity is *O*(*n*).
+    #[unstable(feature = "binary_heap_contains_remove", issue = "82001")]
+    pub fn contains<Q: ?Sized>(&self, item: &Q) -> bool
+    where
+        T: Borrow<Q>,
+        Q: PartialEq,
+    {
+        self.data.iter().any(|x| x.borrow() == item)
+    }
+
+    /// Removes an item from the `BinaryHeap` and returns it, if it exists.
+    ///
+    /// The item may be any borrowed form of the heap's item type, but
+    /// `PartialEq` on the borrowed form *must* match that for the item type. If
+    /// there are multiple matching elements, an arbitrary and unspecified one is
```

Comment:
The language "arbitrary and unspecified" may not be optimal. The thought-of meaning is doubled (btw: if it's arbitrary it can't be unspecified, can it?!) and could suggest that _any_ element (even non-matching ones) are returned. Suggestion: "If there are multiple matching elements, there are no guarantees which of the matching elements is removed on each call to `remove()`."